libs/community/langchain_community/retrievers/document_compressors/base.py

```diff
@@ -0,0 +1,98 @@
+from abc import ABC, abstractmethod
```

Comment:
Without porting this to community, the current implementation breaks the code because there is no "langchain" module in the dependencies for community - which itself is quite strange. 

So here are some of the options to fix this:
 - Port the document compressor base to community just as I did before
 - Move this new feature from community to langchain directly
 - Add langchain as a dependencies for the community module
 
 
![image](https://github.com/langchain-ai/langchain/assets/8368470/b5efb71a-41b3-4305-86ad-20857b823691)

 
 What are you thoughts here? @hwchase17